main.go

```diff
@@ -90,7 +90,14 @@ func main() {
 		ginutil.RecoverProblem,
 	)
 
-	if config.HTTP.CORS.AllowAllOrigins {
+	if len(config.HTTP.CORS.AllowOrigins) > 0 {
+		log.Info().Message("Allowing " + config.HTTP.CORS.AllowOrigins + " origins in CORS.")
+		corsConfig := cors.DefaultConfig()
+		corsConfig.AllowOrigins = []string{config.HTTP.CORS.AllowOrigins}
+		corsConfig.AddAllowHeaders("Authorization")
+		corsConfig.AllowCredentials = true
+		r.Use(cors.New(corsConfig))
```

Comment:
Be careful when merging this so it doesn't overwrite the #101 version of the code.